src/GuppyDataExplorer/GuppyDataExplorer.jsx

```diff
@@ -13,29 +13,98 @@ import {
   ChartConfigType,
 } from './configTypeDef';
 import './GuppyDataExplorer.css';
+import isEnabled from '../helpers/featureFlags';
 
 class GuppyDataExplorer extends React.Component {
   constructor(props) {
     super(props);
+    let initialState = {};
+    let initialFilter = {};
+
+    // aborting URL decode
+    if (isEnabled('storeExplorerStateInURL')) {
+      const stateFromURL = getQueryParameter('filters');
+      if (stateFromURL) {
+        const decodedFilter = atob(stateFromURL);
+        const isValidJSON = IsValidJSONString(decodedFilter);
+        if (isValidJSON) {
+          initialState = JSON.parse(decodedFilter);
+          initialFilter = initialState.filter;
+        }
+      }
+    }
+
     this.state = {
       aggsData: {},
       filter: {},
+      initialFilterFromURL: initialFilter,
+      encodableExplorerStateForURL: { filter: initialFilter },
     };
   }
 
+  getEncodedQueryStateFromExplorer = () => {
+    const stateObj = this.state.encodableExplorerStateForURL;
+    if (this.isExplorerStatePristine(stateObj)) {
+      return '';
+    }
+    const encoded = encodeURIComponent(btoa(JSON.stringify(stateObj)));
```

Comment:
It would be nice if you could put some comments into the changes in this code, like briefly explain why are we using `btoa()` and `atob()` in here

We both know we are doing base64 encoding/decoding ere, but just to share the knowledges